### DISCUSSION TITLE
How can I import a custom shx file?

### DISCUSSION THREAD

**Discussion by mapinxue** [mozman/ezdxf#1187] [Q&A]
Hi, need help.
## Question
I am trying to use `ezdxf.odafc` to open dwg files drawn with autocad and save them as svg files. However, the files contain existing shx font files. How should I import the font files and export them to svg correctly?

## What I have tried
```py
import ezdxf

ezdxf.options.support_dirs = [
    'D:\Fonts' # this Fonts folder has shx files.
]
## ...

 doc.styles.add_shx('font.shx', dxfattribs={'font': 'font.shx'})
```
The code does not work, what can I do to import shx file and use it?

**Comment by mozman** [comment]
1. Most options are only read at startup (support folders, paths to executables), changing these values has no effect at runtime. To change these options, you must create a configuration file, see section [Config Files](https://ezdxf.mozman.at/docs/options.html#config-files).
2. The styles should already be defined in the DXF files (if the DXF file is correct), no need to add these SHX files
3. Adding SHX fonts is done by `doc.styles.add('NEW_STYLE', font='new_font.shx'})`
4. The `add_shx()` method handles is a special case, you don't need.
